cmd/query/app/apiv3/http_gateway.go

```diff
@@ -166,11 +161,18 @@ func (h *HTTPGateway) getTrace(w http.ResponseWriter, r *http.Request) {
 		}
 		request.RawTraces = rawTraces
 	}
-	trc, err := h.QueryService.GetTrace(r.Context(), request)
+	getTracesIter := h.QueryService.GetTraces(r.Context(), request)
+	aggrTracesIter := jptrace.AggregateTraces(getTracesIter)
+	trc, err := iter.CollectWithErrors(aggrTracesIter)
+
 	if h.tryHandleError(w, err, http.StatusInternalServerError) {
 		return
 	}
-	h.returnSpans(trc.Spans, w)
+	if len(trc) == 0 {
+		// TODO: should we return 404 if trace not found?
```

Comment:
yes, I think this is the contract of v3 API (which is unfortunate since it doesn't match the Storage v2 API contract, but we'll keep it for backwards compatibility).